### DISCUSSION TITLE
I solved the problem of horizontal 2.8 inch ST7789

### DISCUSSION THREAD

**Discussion by salarizadi** [Moddable-OpenSource/moddable#1178] [Announcements]
I managed to solve the ST7789 2.8 inch screen setup problems. You may also face my problems, but don't worry, these are already solved. You can use my explanation to solve the problem.

First, let's go to the previous 2 discussions to see what happened:

https://github.com/Moddable-OpenSource/moddable/discussions/1175

https://github.com/Moddable-OpenSource/moddable/discussions/1176

Now let's move on to solving the problem of the touch not working properly in the horizontal mode of the display:

https://github.com/Moddable-OpenSource/moddable/assets/67143370/3080d5aa-8cef-4043-848b-33be13c9572a

I uploaded a similar video in the previous 2 threads, but now you can see the keyboard working correctly in landscape mode.

how did i fix it?

Instead of setting the screen rotation to 90 or 270 in the manifest, I used the command screen.rotation = 270 and the problem was solved.

Good luck all😉

**Comment by phoddie** [comment]
Thanks for sharing your solution! Congratulations on working through this.

FWIW – when you set the rotation in the manifest (or command line), the rotation is applied at build time. When you do it by setting `screen.rotation` it is applied at runtime. You can read more about this on our blog post "[Run-time Display Rotation"](https://www.moddable.com/blog/rotate/). The ILI9341 / ST7789 display driver supports both, but some only support build time rotation
